Задание VI

PrideSt opened the pull request:
- [ ] Нужно реализовать CRUD сервис, для хранения информации для Вашей главной сущности
- [ ] ручки будут CreateEntity , DescribeEntity, ListEntities, RemoveEntity - для синхронного общения (gRPC). Entity - для примера
- [ ] Нужно описать proto файлы и сгенерировать код в pkg директорию
- [ ] Makefile можно посмотреть здесь: https://github.com/zerospiel/dotfiles/blob/81a271153f9ba7aec24cbfafe7fb80b1e8f6647c/example/Makefile
- [ ] Имплементация ручек - просто логгирование с помощью zerolog
- [ ] Добавить в cmd/ova-entity-api/main.go код по поднятию gRPC сервиса

Commit:
issue-6 add gRPC api handlers for joke

Commit:
issue-6 add gRPC api handlers for joke

Commit:
issue-6 fix after refactor, package names of joke and author changed to models

PrideSt commented:
Здравствуйте, проверьте пожалуйста домашнее задание. Изменения не связанные с вендорингом gRPC можно посмотреть тут https://github.com/ozonva/ova-joke-api/compare/22cb221...20545c4

dmzkrsk commented:
MR большой, посмотрю потихоньку

Commit:
issue-6 change api, replace joke.Author from object to uint64

dmzkrsk commented on api/ova-joke-api/ova-joke-api.proto:
```diff
@@ -0,0 +1,63 @@
+// JokeAPI service for working with jokes.
+service JokeService {
+  // CreateJokeV1 create new joke entity.
+  rpc CreateJokeV1(CreateJokeRequestV1) returns (CreateJokeResponseV1) {
```
По опыту, в Озоне `V1` не пишем

PrideSt replied:
Удалил суффикс V1, переименовал файлы с API методами сервиса

dmzkrsk commented on internal/ova-joke-api/describe_v1.go:
```diff
@@ -0,0 +1,41 @@
+
+	if !ok {
+		msg := fmt.Sprintf("joke with id=%d not found", req.Id)
+		log.Error().Msg(fmt.Sprintf("describe: %s", msg))
```
кажется, это не ошибка, а рабочее поведение

dmzkrsk replied:
Тут надо себе задать вопрос, что ты будешь делать, если увидишь такую ошибку в логах? Если ничего, то лог не нужен

PrideSt replied:
Заменил на warning, в случае большого кол-ва таких ошибок можно обратить внимание и понять что кто-то некорректно пользуется предоставленным API либо есть какие-то нарушения согласованности. Конечно ошибки для такого многовато

dmzkrsk commented on internal/ova-joke-api/remove_v1.go:
```diff
@@ -0,0 +1,32 @@
+
+	if _, ok := j.jokes.data[req.GetId()]; !ok {
+		msg := fmt.Sprintf("joke with id=%d not found", req.Id)
+		log.Error().Msg(fmt.Sprintf("remove: %s", msg))
```
Тут тоже можно подумать над критичностью

PrideSt replied:
Заменил ошибку в логировании на warning, ошибку заменил на nil.
Но в реализации с использованием БД в 7-й домашке, оставил возврат ошибки из метода (+error), потому что причиной будет являться нарушение работы с БД, ошибка отсутствия результата обрабатывается на уровне репозитория.

Commit:
issue-6 fix rewiever comments

remove V1 suffix from gRPC request, response
change some errors in handlers logging to warnings
RemoveJoke now return nil instead error when entity
with given id not exists

PrideSt commented:
@dmzkrsk посмотри пожалуйста правки https://github.com/ozonva/ova-joke-api/pull/10/commits/f3c2e05a8c4cb4cbd4460aa948bae69f27055aec
